cocotb/scheduler.py

```diff
@@ -512,7 +512,10 @@ def unschedule(self, coro):
                 self._test._force_outcome(outcome)
             except Exception as e:
                 if _pdb_on_exception:
-                    pdb.post_mortem(e.__traceback__)
+                    if not hasattr(e, '__traceback__'):
```

Comment:
The [documentation](https://docs.python.org/2/library/pdb.html#pdb.post_mortem) seems to indicate that you can call `post_mortem` without an argument to use the currently handled exception. And that would work on Python 2 and Python 3.

Reply:
Nice!  That's much better.

I've also noticed that deals with the Exceptions here doesn't work when the exception was raised in the cocotb.test function or in any function that the test is directly awaiting.  It only works for explicitly forked coroutines so I also need to fix that.

Reply:
Yes, the main test will never be `unschedule`d if it fails, the exception will trickle back up to the regression manager. You'll probably want to move this code in `RegressionManager.handle_result`.

Reply:
@ktbarrett is right in both of their comments :)